### PR TITLE
Skyline: Add support for precursor isotopes and reporter ions to transition list import

### PR DESCRIPTION
**Pull request by brendanx67**
- reporter ions was requested by Thermo for TMT support
- others have requested support for the MS1 precursor isotopes supported by the Targets list

### CODE REVIEW

**Review by nickshulman — APPROVED** [review]
Looks good to me.

File: pwiz_tools/Skyline/Model/Import.cs [hunk 1]
```diff
@@ -1314,6 +1320,67 @@ private ExTransitionInfo CalcTransitionExplanations(ExTransitionInfo info, strin
...
+
+            /// <summary>
+            /// Finds the closest custom product ion if any to a specified product m/z value.
+            /// </summary>
+            private ProductExp GetCustomIonProductExp(double productMz, ExTransitionInfo info)
+            {
+                double minDeltaMz = double.MaxValue;
+                ProductExp productExp = null;
+                var massType = Settings.TransitionSettings.Prediction.FragmentMassType;
+
```
Comment (bspratt) [inline pwiz_tools/Skyline/Model/Import.cs:1367]: Is it always the case that we aren't provided at least a charge to go with the m/z? That could help disambiguate, when/if available.

Comment (brendanx67) [reply pwiz_tools/Skyline/Model/Import.cs:1367]: Right. In proteomics, we are not guaranteed the user supplied a charge. I think the minimum information is sequence, precursor m/z, and product m/z.

Comment (bspratt) [reply pwiz_tools/Skyline/Model/Import.cs:1367]: That's what I figured, but it seems like we ought to consult it when it is provided?

Comment (brendanx67) [reply pwiz_tools/Skyline/Model/Import.cs:1367]: Good point. I did that for the precursor charge, but not the custom ion charge. Good catch.

**Comment by brendanx67** [comment]
@nickshulman @bspratt When I started messing with the first transition in my test list, I realized things were a bit more complicated than I had thought. The old code failed if you started the list with a non-monoisotopic precursor or a custom ion. That the monoisotopic precursor worked was a bit of a fluke, just because it is valid for both MS1 and MS/MS spectra. I added code to fix these cases and tests to make sure they work.

**Pull request merged**